src/libxtp/threecenter_gwbse.cc

```diff
@@ -67,6 +86,11 @@ void TCMatrix_gwbse::Fill(const AOBasis& gwbasis, const AOBasis& dftbasis,
   _dftbasis = &dftbasis;
   _dft_orbitals = &dft_orbitals;
 
+  // If cuda is enabled the dft orbitals are sent first to the cuda device
+#if defined(USE_CUDA)
+  auto cuda_matrices = SendDFTMatricesToGPU(dft_orbitals);
+#endif
+
   // loop over all shells in the GW basis and get _Mmn for that shell
 #pragma omp parallel for schedule(guided)  // private(_block)
```

Comment:
this has to be dynamic now, afaik, not guided, because the CUDA supported threads should be much faster than the rest